crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,430 @@
+//! Components used to control the layout of UI-[`Node`](super::Node) entities.
+use crate::{Size, UiRect, Val};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_reflect::prelude::*;
+use serde::{Deserialize, Serialize};
+
+use self::flex::FlexLayout;
+
+/// Controls which layout algorithm is used to position this UI node
+#[derive(Component, Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum Layout {
+    /// Use the absolute position and size specified.
+    PositionLayout,
+    /// Use the [Flexbox](https://cssreference.io/flexbox/) layout algorithm.
+    FlexLayout(FlexLayout),
+}
+
+impl Default for Layout {
+    fn default() -> Self {
+        Layout::FlexLayout(FlexLayout::default())
+    }
+}
+
+/// The strategy used to position this node.
+#[derive(
+    Component, Copy, Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize, Reflect,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum PositionType {
+    /// Relative to all other nodes with the [`PositionType::Relative`] value.
+    #[default]
+    Relative,
+    /// Positioned in relation to its parent,
```

Comment:
I'm also not sure it's true. At least in CSS pre-flexbox this is wrong, the absolute child is positioned relative to the innermost parent, ignoring all relative ancestors. I couldn't find an info for flexbox, this doesn't seem to be part of the flexbox model.

Reply:
Here's what I could find, but honestly I don't really understand it:

[MDN - `position: absolute`](https://developer.mozilla.org/en-US/docs/Web/CSS/position#values):

> The element is removed from the normal document flow, and no space is created for the element in the page layout. It is positioned relative to its closest positioned ancestor, if any; otherwise, it is placed relative to the initial [containing block](https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_block). Its final position is determined by the values of top, right, bottom, and left.

[MDN - Identifying the containing block](https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block):

> If the position property is `absolute`, the containing block is formed by the edge of the padding box of the nearest ancestor element that has a `position` value other than `static` (`fixed`, `absolute`, `relative`, or `sticky`).

In any case, I think the docs are correct with regards to how `taffy` currently behaves (to my knowledge).

If anyone wants to confirm, this should be the relevant code:

<https://github.com/DioxusLabs/taffy/blob/d295d9b5a1de28fa6644a7a4d5fefcd3b6957773/src/flexbox.rs#L1455-L1609>